tests/test_project.py

```diff
@@ -703,6 +703,26 @@ def op2(job):
         self.mock_project(project_class=B).detect_operation_graph()
 
 
+class TestProcessProject(TestProject):
```

Comment:
These classes seem a bit excessive. Could we not just have test functions in `TestProject` that just used the `config_overrides` in the `mock_project` function?